h/services/profile_group.py

```diff
@@ -25,17 +30,32 @@ def all(self, user=None, authority=None):
         if user is not None:
             private_groups = user.groups
 
-        groups = open_groups + private_groups
+        groups = self.sort(open_groups) + self.sort(private_groups)
 
         return [self._group_model(group) for group in groups]
 
     def _group_model(self, group):
-        model = {'name': group.name, 'id': group.pubid, 'public': group.is_public}
+        model = {
+          'name': group.name,
+          'id': group.pubid,
+          'public': group.is_public,
+          'scoped': False,  # TODO
+          'type': 'open' if group.is_public else 'private',  # TODO
```

Comment:
`'open' if group.is_public else 'private'` is true for now but I guess it might not be at some point in future (e.g. once restricted groups exist). At some point I guess this might call a computed `type` property on the group object itself, once we have one. In the meantime this is a nice way for API work to not be blocked by model and service work.

Reply:
That's why the `# TODO` :) — I assume this is going to change—and, yah, I was aiming not to have too much coupling between what I'm doing and what you're doing!